### PR TITLE
statesync: convert apphash to hex string in log

### PR DESCRIPTION
**Pull request by JayT106**
noticed this place is still printing unreadable apphash, convert to hex string

---

#### PR checklist

- [ ] Tests written/updated, or no tests needed
- [ ] `CHANGELOG_PENDING.md` updated, or no changelog entry needed
- [ ] Updated relevant documentation (`docs/`) and code comments, or no
      documentation updates needed

### CODE REVIEW

File: statesync/syncer.go [hunk 1]
```diff
@@ -504,6 +504,6 @@ func (s *syncer) verifyApp(snapshot *snapshot, appVersion uint64) error {
 		return errVerifyFailed
 	}
 
-	s.logger.Info("Verified ABCI app", "height", snapshot.Height, "appHash", snapshot.trustedAppHash)
+	s.logger.Info("Verified ABCI app", "height", snapshot.Height, "appHash", log.NewLazySprintf("%X", snapshot.trustedAppHash))
```
Comment (sergio-mena) [inline statesync/syncer.go:507]: Thanks for this.
Would you mind replacing line 494 by:

```go
"expected", fmt.Sprintf("%X", snapshot.trustedAppHash),
```
(no need for lazy formatter as it is an error condition, not a common path)

Comment (JayT106) [reply statesync/syncer.go]: updated!

Comment (sergio-mena) [reply statesync/syncer.go]: Thanks!

**Review by sergio-mena — APPROVED** [review]

**Review by cmwaters — CHANGES_REQUESTED** [review]
Can we use the `LazySprintf` method used in https://github.com/tendermint/tendermint/pull/9471. Otherwise looks good

**Comment by JayT106** [comment]
> Can we use the `LazySprintf` method used in #9471. Otherwise looks good

https://github.com/tendermint/tendermint/pull/9591#discussion_r999260456
I am good in both ways, any rule we should follow in the future?

**Comment by sergio-mena** [comment]
> > Can we use the `LazySprintf` method used in #9471. Otherwise looks good
> 
> [#9591 (comment)](https://github.com/tendermint/tendermint/pull/9591#discussion_r999260456) I am good in both ways, any rule we should follow in the future?

The rule of thumb is to use the `LazySprintf` in positive paths (those that are hit often), and doing without in error paths, because `LazySprintf` has some implications on the variables passed to it, so we don't want to use it where it is not needed.

File: consensus/replay.go [hunk 2]
```diff
@@ -254,7 +254,7 @@ func (h *Handshaker) Handshake(proxyApp proxy.AppConns) error {
 
 	h.logger.Info("ABCI Handshake App Info",
 		"height", blockHeight,
-		"hash", appHash,
+		"hash", fmt.Sprintf("%X", appHash),
```
[the remark this answers is not in the document]
Comment (sergio-mena) [reply consensus/replay.go:257]: You're right. This is a "positive path".
@JayT106 would you mind using the LazySprintf here?

**Review by cmwaters — APPROVED** [review]

File: consensus/replay.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmwaters) [reply consensus/replay.go]: In the interest of speeding this up, I've just done the change. Let me know if anything else is needed @sergio-mena else you can merge it

[the remark this answers is not in the document]
Comment (JayT106) [reply consensus/replay.go]: thanks @cmwaters @sergio-mena !

[the remark this answers is not in the document]
Comment (sergio-mena) [reply consensus/replay.go]: thanks @cmwaters, LGTM

**Review by thanethomson — APPROVED** [review]

**Comment by sergio-mena** [comment]
Thanks @thanethomson , I updated the branch to merge it several times, but then always got distracted with something else

**Pull request merged**